flink-python/src/main/java/org/apache/flink/client/python/PythonGatewayServer.java

```diff
@@ -70,11 +70,16 @@ public static void main(String[] args) throws IOException {
 			System.exit(1);
 		}
 
-		// Exit on EOF or broken pipe.  This ensures that the server dies
-		// if its parent program dies.
-		while (System.in.read() != -1) {
-			// Do nothing
+		try {
+			// Exit on EOF or broken pipe.  This ensures that the server dies
+			// if its parent program dies.
+			while (System.in.read() != -1) {
+				// Do nothing
+			}
+			gatewayServer.shutdown();
+			System.exit(0);
+		} catch (IOException e) {
+			System.exit(1);
```

Comment:
Maybe we should add this exit in the finally block and also print the error message of the error.